File: src/subside/plugins/koth/commands/CommandAsMember.java
package subside.plugins.koth.commands;

import java.util.ArrayList;
import java.util.List;

import org.bukkit.command.CommandSender;

import subside.plugins.koth.commands.CommandHandler.CommandCategory;
import subside.plugins.koth.utils.IPerm;
import subside.plugins.koth.utils.MessageBuilder;
import subside.plugins.koth.utils.Perm;

public class CommandAsMember extends AbstractCommand {

    public CommandAsMember(CommandCategory category) {
        super(category);
    }

    @Override
    public void run(CommandSender sender, String[] args) {
        List<String> list = getPlugin().getConfigHandler().getGlobal().getHelpCommand();
        List<String> list2 = new ArrayList<>();
        for (String hlp : list) {
           MessageBuilder mB = new MessageBuilder(hlp);
           try {
               mB.koth(getPlugin().getKothHandler().getRunningKoth().getKoth());
               mB.time(getPlugin().getKothHandler().getRunningKoth().getTimeObject());
           }
           catch (Exception e) {
               mB.koth(getPlugin().getKothHandler(), "None").time("00:00").capper("None");
           }
           list2.addAll(mB.buildArray());
       }
       sender.sendMessage(list2.toArray(new String[list2.size()]));
   }

    @Override
    public IPerm getPermission() {
        return Perm.HELP;
    }

    @Override
    public String[] getCommands() {
        return new String[]{"asmember"};
    }

    @Override
    public String getUsage() {
        return "/koth asmember";
    }

    @Override
    public String getDescription() {
        return "Shows the help menu as a normal player";
    }

}

File: src/subside/plugins/koth/commands/CommandAsMember.kt
package subside.plugins.koth.commands

import org.bukkit.command.CommandSender
import subside.plugins.koth.KothManager
import subside.plugins.koth.utils.IPerm
import subside.plugins.koth.utils.MessageBuilder
import subside.plugins.koth.utils.Perm
import java.util.ArrayList

class CommandAsMember : Command {
    override fun run(kothManager: KothManager, sender: CommandSender, args: Array<String>) {
        val rKoth = kothManager.kothHandler.runningKoth
        val list = kothManager.configHandler.global.helpCommand.map { help ->
            MessageBuilder(help)
                .koth(kothManager.kothHandler, rKoth?.koth ?: "None")
                .let { if (rKoth != null) it.time(rKoth.timeObject) else it.time("00:00")}
                .capper(rKoth?.capper?.name ?: "None")

        }
        sender.sendMessage(list.toTypedArray())
    }

    override val permission: IPerm = Perm.HELP
    override val commands: Array<String> = arrayOf("asmember")
    override val usage: String = "/koth asmember"
    override val description: String = "Shows the help menu as a normal player"
}